src/Common/Core/Impl/Extensions/IOExtensions.cs

```diff
@@ -100,8 +100,10 @@ public static class IOExtensions {
 
         public static bool HasReadPermissions(this string path) {
             try {
-                Directory.GetFiles(path);
-                return true;
+                if (Directory.Exists(path)) {
```

Comment:
Really the issue is that we store folder as a singleton but really should be storing per connection